qa/rpc-tests/sc_cert_base.py

```diff
@@ -63,36 +63,40 @@ def run_test(self):
         blocks.extend(self.nodes[0].generate(220))
         self.sync_all()
 
-        bal_before = self.nodes[1].getbalance("", 0)
-        mark_logs("Node1 balance before SC creation: {}".format(bal_before), self.nodes, DEBUG_MODE)
-
+        # SC creation
+        bal_before_sc_creation = self.nodes[1].getbalance("", 0)
+        mark_logs("Node1 balance before SC creation: {}".format(bal_before_sc_creation), self.nodes, DEBUG_MODE)
         creating_tx = self.nodes[1].sc_create(scid, EPOCH_LENGTH, "dada", creation_amount, "abcdef")
         mark_logs("Node 1 created the SC spending {} coins via tx {}.".format(creation_amount, creating_tx), self.nodes, DEBUG_MODE)
         self.sync_all()
 
         mark_logs("Node0 confirms Sc creation generating 1 block", self.nodes, DEBUG_MODE)
         blocks.extend(self.nodes[0].generate(1))
-        sc_creating_height = self.nodes[0].getblockcount()  # Should not this be in SC info??'
+        sc_creating_height = self.nodes[0].getblockcount()
         self.sync_all()
 
-        # fee can be seen on sender wallet (it is a negative value)
+        # Check node 1 balance following sc creation
         fee = self.nodes[1].gettransaction(creating_tx)['fee']
-        mark_logs("Fee paid for SC creation: {}".format(fee), self.nodes, DEBUG_MODE)
+        mark_logs("Fee paid for SC creation: {}".format(fee), self.nodes, DEBUG_MODE)        
+        bal_after_sc_creation = self.nodes[1].getbalance("", 0)
+        mark_logs("Node1 balance after SC creation: {}".format(bal_after_sc_creation), self.nodes, DEBUG_MODE)
+        assert_equal(bal_before_sc_creation, bal_after_sc_creation + creation_amount - fee)
 
-        # node 1 has just the coinbase minus the sc creation amount
-        assert_equal(bal_before, self.nodes[1].getbalance("", 0) + creation_amount - fee)
-        mark_logs("Node1 balance after SC creation: {}".format(self.nodes[1].getbalance("", 0)), self.nodes, DEBUG_MODE)
 
+        # Fwd Transfer to Sc
+        bal_before_fwd_tx = self.nodes[0].getbalance("", 0)
+        mark_logs("Node0 balance before fwd tx: {}".format(bal_before_fwd_tx), self.nodes, DEBUG_MODE)
         fwd_tx = self.nodes[0].sc_send("abcd", fwt_amount, scid)
-        mark_logs("Node 0 transfers {} coins to SC with tx {}...".format(fwt_amount, fwd_tx), self.nodes, DEBUG_MODE)
+        mark_logs("Node0 transfers {} coins to SC with tx {}...".format(fwt_amount, fwd_tx), self.nodes, DEBUG_MODE)
         self.sync_all()
 
         mark_logs("Node0 confirms fwd transfer generating 1 block", self.nodes, DEBUG_MODE)
         blocks.extend(self.nodes[0].generate(1))
         self.sync_all()
 
+        # TODO: Check node0 balance following fwd transfert
```

Comment:
This is marked as TODO and should be added

Reply:
Agree. Indeed these extra checks have been precious as they uncovered faulty handling of some corner cases